app/src/gui/components/workspace/create_new_survey.tsx

```diff
@@ -0,0 +1,436 @@
+  const [selectedTemplate, setSelectedTemplate] = useState<string>('');
+  const [openDialog, setOpenDialog] = useState(false);
+  const [surveyName, setSurveyName] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+  const navigate = useNavigate();
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+
+  const getListingFromDB = async (): Promise<ListingsObject | undefined> => {
+    try {
+      const allListings = await directory_db.local.allDocs({
+        include_docs: true,
+      });
+      if (allListings.rows.length > 0) {
+        return allListings.rows[0].doc as ListingsObject;
+      }
+    } catch (err) {
+      console.error('Error fetching listing from the database', err);
+    }
+    return undefined;
+  };
+
+  const loadTemplates = async () => {
+    try {
+      const listing = await getListingFromDB();
+      console.log('listing in loadTemp;ates', listing);
+      if (listing) {
+        const response = await fetchTemplates(listing);
+        console.log('loadtemplates', response);
+        if (response && response.templates) {
+          console.log('response for templates', response.templates);
+          setTemplates(response.templates);
+        } else {
+          console.error('No templates found or error occurred.');
+        }
+      } else {
+        console.error('No listing found.');
+      }
+    } catch (error) {
+      console.error('Error fetching templates:', error);
+    }
+  };
+
+  useEffect(() => {
+    loadTemplates();
+  }, []);
+
+  useEffect(() => {
+    const originalStyle = window.getComputedStyle(document.body).paddingRight;
+
+    if (openDialog) {
+      document.body.style.paddingRight = '0px';
+    }
+
+    return () => {
+      document.body.style.paddingRight = originalStyle;
+    };
+  }, [openDialog]);
+
+  const handleTemplateChange = (event: SelectChangeEvent<string>) => {
+    const templateId = event.target.value;
+    console.log('Template selected:', templateId);
+    setSelectedTemplate(templateId);
+  };
+
+  const handleInstantiateSurvey = () => {
+    if (!selectedTemplate) {
+      setError('Please select a template before creating a survey');
+      return;
+    }
+    setError(null);
+    console.log('Instantiate Survey button clicked');
+    setOpenDialog(true);
+  };
+
+  const handleCloseDialog = () => {
+    console.log('Dialog closed');
+    setOpenDialog(false);
+  };
+
+  // Handle survey creation
+  const handleSubmitSurvey = async () => {
+    if (!surveyName) {
+      setError('Survey Name is required');
+      console.error('Survey Name is missing');
+      return;
+    }
+    if (!selectedTemplate) {
+      setError('Please select a template');
+      console.error('Template is not selected');
+      return;
+    }
+    console.log('RANIOIIII Starting survey creation with:', {
```

Comment:
Left over?

Reply:
Ahh no. Sorry about the logs. I thought I removed them but damnnn!!